### PR TITLE
Add an INJECT flag, that can be used with keyswitch states

### PR DESCRIPTION
**Pull request by algernon**
This flag can be used by the different key event handlers to decide whether the event is a result of a direct key press, or if it was injected into the event loop. This is very useful for those handlers that override existing keycodes, and change their behaviour (such as SpaceCadet Shift, and one-shot modifiers in auto mode), since they don't have to play games to avoid recursion, or to avoid changing injected events.

As an additional fun fact, checking bits instead of truthiness seems to result in smaller code.

### CODE REVIEW

File: src/keyswitch_state.cpp [hunk 1]
```diff
@@ -11,7 +11,7 @@ boolean key_was_pressed (byte keyState) {
 }
 
 boolean key_was_pressed (uint8_t currentState, uint8_t previousState) {
-    if (previousState) {
+    if (previousState & B00000001) {
```
Comment (TimNN) [inline src/keyswitch_state.cpp]: Any reason this (and the ones below) are not using the `INJECTED` define?  That would seem clearer to me.

Comment (algernon) [inline src/keyswitch_state.cpp]: You are right though, that a constant for these would be clearer. Will push another commit, thanks for the hint!

**Comment by obra** [comment]
We may at some point want to reconsider the 8 bit vector for keypress state. It was originally there because it was how we were doing debouncing. Now that we're doing debouncing elsewhere, we're using it somewhat differentl and the same structure may not make sense at all.

**Comment by algernon** [comment]
I can have a go at collapsing the `currentState`/`previousState` stuff into a single byte. These two would be one bit each, `INJECTED` a third, and then we have a few more for additional info.

**Comment by algernon** [comment]
First, I wanted to edit this, but... I'll rename the branch too, as it no longer reflects the contents.

**Pull request closed without merging**